trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
```

Comment:
.....okay, I just re-read PEP 479 and this is more complicated than I realized. I'm actually not sure that what I said above about replacing `StopIteration` with `StopAsyncIteration` is correct, and actually it's quite likely that `async_generator` doesn't handle this correctly. Need to do some experiments to see exactly what the python 3.6 native async generator behavior actually is...